packages/tracing/src/idletransaction.ts

```diff
@@ -0,0 +1,284 @@
+// tslint:disable: max-classes-per-file
+import { Hub } from '@sentry/hub';
+import { TransactionContext } from '@sentry/types';
+import { logger, timestampWithMs } from '@sentry/utils';
+
+import { Span } from './span';
+import { SpanStatus } from './spanstatus';
+import { SpanRecorder, Transaction } from './transaction';
+
+const DEFAULT_IDLE_TIMEOUT = 1000;
+
+/**
+ * @inheritDoc
+ */
+export class IdleTransactionSpanRecorder extends SpanRecorder {
+  private readonly _pushActivity: (id: string) => void;
+  private readonly _popActivity: (id: string) => void;
+  public transactionSpanId: string = '';
+
+  public constructor(
+    pushActivity: (id: string) => void,
+    popActivity: (id: string) => void,
+    transactionSpanId: string = '',
+    maxlen?: number,
+  ) {
```

Comment:
Because you initialize them here with defaults, you can drop L16-18 and move these prefix declarations here.

```suggestion
  public constructor(
    private readonly pushActivity: (id: string) => void,
    private readonly popActivity: (id: string) => void,
    public transactionSpanId: string = '',
    maxlen?: number,
  ) {
```

Reply:
TIL. changed this and one below.